### PR TITLE
Hotkeys for Modded Lenses

### PR DESCRIPTION
**Pull request by the-m4a**
Requsted in #320
Adds configurable hotkeys for the modded lenses: Archaeologist, Barbarian, Builder, City Overlap, Naturalist, Resource, Routes, Wonder, and Scout.  The hot keys are configured in the Civ6 standard key bindings menu.  They work like the current lens hot keys (pressing "1" for the religion lens).  By default I assigned them all to SHIFT+someNumber.

I did look into moving the CQUI hot keys here, but we run into an issue where a single key is assigned to multiple actions (like "R" or "H"), and the KeyBindings UI does not allow you to do that.  A bunch of CQUI things ended up blank, so I abandoned that effort.

image

### CODE REVIEW

**Comment by RustyD** [comment]
I checked out your branch and had a very quick test - the Resource Lens  :-)
When I go into the Key Bindings options directly from the Main Menu, the text description (of the new lens bindings) are not filled in. This is on Linux. When I go into the Key Bindings options from within a game, everything seems to be good, as shown in your picture above.

Hitting key '4' in the game did display the resource lens, so that seems to be working fine.

Civ6_MM_KeyBindings

Thanks for working on this so quick  :-)

**Comment by the-m4a** [comment]
That's weird.  Do the text strings for the lenses themselves (in the lenses list above the minimap) show up okay?  Actually - did you load this branch after starting Civ6?  Maybe some goofiness with loading of the text strings

**Comment by the-m4a** [comment]
also no prob.  the hot keys stuff was something i had been curious about.  bummer that there is not a way that i could see to assign the same key to multiple commands.  i suppose it makes sense, but still

**Comment by RustyD** [comment]
Yeah, the text strings for the lenses (in the list above the minimap) are fine. I switched to your branch before loading Civ (or even Steam), and have done so a few times with the same result.

For me, this is not a big deal. I just thought I'd draw your attention to it before you merge, in case there was a simple/obvious solution. From my perspective, I don't think it should stop you from releasing this useful update.

**Comment by the-m4a** [comment]
it being okay in one place and not okay in the other is a weird issue, since it refers to the same file.  maybe it's the point at which the file loads ("FrontEndActions" instead of "InGameActions").  I have added the list of MoreLenses text files to the FrontEndActions part as well - maybe they'll load now?

Also - @alexeyOnGitHub or @Infixo can you review/approve if it looks good

**Comment by RustyD** [comment]
Yes - that's fixed it.

A very quick check of the Key Bindings off the main menu and also in game, and they look to both be OK.

**Comment by the-m4a** [comment]
@alexeyOnGitHub - hello, I'm finally back from various not-being-home over the last few months.  May start doing these things again.  Can you approve this PR?  Should be good to go

**Review by RustyD — APPROVED** [review]

**Pull request merged**
